crates/rpc-types-eth/src/transaction/mod.rs

```diff
@@ -80,6 +79,26 @@ where
     }
 }
 
+impl<T> Transaction<T>
+where
+    T: TransactionTrait + Encodable2718,
+{
+    /// Returns the [`TransactionInfo`] for this transaction.
+    ///
+    /// This contains various metadata about the transaction and block context if available.
+    pub fn info(&self) -> TransactionInfo {
+        TransactionInfo {
+            hash: Some(self.tx_hash()),
+            index: self.transaction_index,
+            block_hash: self.block_hash,
+            block_number: self.block_number,
+            // We don't know the base fee of the block when we're constructing this from
+            // `Transaction`
+            base_fee: None,
```

Comment:
should it take `base_fee` as an argument?

Reply:
we set this via the additional builder fn: `into().with_base_fee()`